keras_cv/layers/preprocessing/channel_shuffle.py

```diff
@@ -65,10 +72,10 @@ def augment_image(self, image, transformation=None):
             )
 
         channels_per_group = num_channels // self.groups
-        image = tf.reshape(image, [height, width, self.groups, channels_per_group])
-        image = tf.transpose(image, perm=[2, 0, 1, 3])
-        image = tf.random.shuffle(image, seed=self.seed)
-        image = tf.transpose(image, perm=[1, 2, 3, 0])
+
+        rand_indices = tf.argsort(self.rand_uniform(shape=[self.groups]))
```

Comment:
@LukeWood cc. @bhack 
I've noticed that, the `tf.argsort` fallback to while loop here. 

```
Tensor("loop_body/argsort/TopKV2:1", shape=(3,), dtype=int32)
WARNING:tensorflow:Using a while_loop for converting TopKV2
```

Reply:
> @LukeWood cc. @bhack I've noticed that, the `tf.argsort` fallback to while loop here.
> 
> ```
> Tensor("loop_body/argsort/TopKV2:1", shape=(3,), dtype=int32)
> WARNING:tensorflow:Using a while_loop for converting TopKV2
> ```

Just another one to add to the list. But as you can see with the master implementation we have also `RandomShuffle` in Keras-cv master (one in. one out):
https://github.com/keras-team/keras-cv/issues/291#issuecomment-1112210556